### ISSUE TITLE
Deleting the root of an SPP model crashes OMEdit

### ISSUE THREAD

**Issue by perost** [OpenModelica/OpenModelica#13089]
### Description

Deleting the root system of an SPP model causes OMEdit to immediately crash with a segmentation fault.

### Steps to Reproduce

1. Create a new SPP model.
2. Go to the diagram view of the new model and delete the Root node.
3. OMEdit segfaults when creating the undo command.

### Expected Behavior

I'm not sure what the expected behaviour actually is, except to not crash.

### Version and OS

 - OpenModelica Version: master
 - OS: Linux

### Additional Context

This might be the cause of some crash reports we've gotten lately.

**Comment by bilderbuchi** [comment]
I can reproduce on Windows, partial backtrace on Windows (the crash apparently took my gdb down with it), but maybe it helps:
```
(gdb) bt
#0  0x00007ffed6415f10 in ucrtbase!strlen () from C:\Windows\System32\ucrtbase.dll
#1  0x00007ff79fc65253 in QString::QString (this=0x5fbb78,
    ch=0xabababababababab <error: Cannot access memory at address 0xabababababababab>)
    at C:/OMDevUCRT/tools/msys/ucrt64/include/QtCore/qstring.h:836
#2  0x00007ff79f770b65 in LibraryTreeModel::createLibraryTreeItems (this=0x27e94500, pLibraryTreeItem=0x2916f980)
    at C:/Repository/OM_source/OMEdit/OMEditLIB/Modeling/LibraryTreeWidget.cpp:2480
#3  0x00007ff79f76c5cf in LibraryTreeModel::createLibraryTreeItem (this=0x27e94500, name=..., nameStructure=..., path=...,
    isSaved=false, pParentLibraryTreeItem=0x27e90780, pOMSElement=0x0, pOMSConnector=0x0, pOMSBusConnector=0x0,
    pOMSTLMBusConnector=0x0, row=5) at C:/Repository/OM_source/OMEdit/OMEditLIB/Modeling/LibraryTreeWidget.cpp:1630
#4  0x00007ff79f76f1d6 in LibraryTreeModel::reLoadOMSimulatorModel (this=0x27e94500, modelName=..., editedCref=..., snapShot=...,
    oldEditedCref=..., newEditedCref=...) at C:/Repository/OM_source/OMEdit/OMEditLIB/Modeling/LibraryTreeWidget.cpp:2155
#5  0x00007ff79fb06b64 in OMSimulatorUndoCommand::redoInternal (this=0x27dcfc80)
    at C:/Repository/OM_source/OMEdit/OMEditLIB/Modeling/Commands.cpp:1521
#6  0x00007ff79f7b3e9d in UndoStack::push (this=0x291ecde0, cmd=0x27dcfc80)
    at C:/Repository/OM_source/OMEdit/OMEditLIB/Modeling/ModelWidgetContainer.cpp:5737
#7  0x00007ff79f7c7c2c in ModelWidget::createOMSimulatorUndoCommand (this=0x2d2d7240, commandText=..., doSnapShot=true,
    switchToEdited=true, oldEditedCref=..., newEditedCref=...)
    at C:/Repository/OM_source/OMEdit/OMEditLIB/Modeling/ModelWidgetContainer.cpp:8140
#8  0x00007ff79f7c61a6 in ModelWidget::endMacro (this=0x2d2d7240)
    at C:/Repository/OM_source/OMEdit/OMEditLIB/Modeling/ModelWidgetContainer.cpp:7929
#9  0x00007ff79f7a956a in GraphicsView::deleteItems (this=0x2d2d8030)
    at C:/Repository/OM_source/OMEdit/OMEditLIB/Modeling/ModelWidgetContainer.cpp:4355
#10 0x00007ff79f7ad8f0 in GraphicsView::keyPressEvent (this=0x2d2d8030,
```

**Comment by adeas31** [comment]
Fixed and ported to maintenance/v1.24

 - https://github.com/OpenModelica/OMSimulator/pull/1324
 - #13094
 - #13100

**Comment by bilderbuchi** [comment]
Confirmed on my Windows, this does not crash anymore.
